server/errorhandler.go

```diff
@@ -0,0 +1,40 @@
+package server
+
+import (
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/containous/traefik/middlewares"
+)
+
```

Comment:
Hmm it has none of the capabilities the `Stateful` interface expects. Maybe the naming is rather not perfect then? It is not an extension of `ResponseWriter` or whatsoever, its an implementation of the oxy `ErrorHandler` interface. Its called `Recording` because it calls the `NetErrorRecorder` which in our case will manipulate the request context. So I think this is unnecessary here. Any suggestion for naming improvements?